wallet/wallet.go

```diff
@@ -606,6 +606,7 @@ out:
 			break out
 
 		case <-w.lockRequests:
+			timeout = nil
```

Comment:
This is going to be set to nil anyways below, but as discussed, I believe the order could be the issue since channel selection is not deterministic.